shared/macros-bash.jinja

```diff
@@ -1769,3 +1769,69 @@ while IFS= read -r {{{ varname }}}; do
     {{{ caller() | indent(4) }}}
 done <   <({{{ command_and_its_args }}})
 {{%- endmacro %}}
+
+
+{{#
+  # This macro is DEPRECATED and should not be used in new remediations, as it does not remediate
+  # config files in the /etc/sssd/conf.d directory
+  #
+  # Try find SSSD_SECT section and SSSD_OPT setting in SSSD_CONF SSSD config file,
+  # if SSSD_OPT option exists, set it to SSSD_OPT_VAL, if SSSD_OPT
+  # doesn't exist add it, if SSSD_SECT section doesn't exist add both
+  # the SSSD_SECT section and the SSSD_OPT configuration option.
+
+:param SSSD_SECT:       section of {{{ SSSD_CONF }}} under which we are looking for {{{ SSSD_OPT }}}}
+:param SSSD_CONF:       the SSSD config file
+:param SSSD_OPT:        the SSSD option we are setting
+:param SSSD_OPT_VAL:    the value of SSSD_OPT we are setting
+
+#}}
+{{% macro bash_sssd_set_option(SSSD_SECT, SSSD_CONF, SSSD_OPT, SSSD_OPT_VAL) -%}}
+# find option in section and change value
+if grep -qzosP "[[:space:]]*{{{ SSSD_SECT }}}([^\n\[]*\n+)+?[[:space:]]*{{{ SSSD_OPT }}}" {{{ SSSD_CONF }}}; then
+        sed -i "s/{{{ SSSD_OPT }}}[^(\n)]*/{{{ SSSD_OPT }}} = {{{ SSSD_OPT_VAL }}}/" {{{ SSSD_CONF }}}
+# find section and add option to it
+elif grep -qs "[[:space:]]*{{{ SSSD_SECT }}}" {{{ SSSD_CONF }}}; then
+        sed -i "/[[:space:]]*{{{ SSSD_SECT }}}/a {{{ SSSD_OPT }}} = {{{ SSSD_OPT_VAL }}}" {{{ SSSD_CONF }}}
+else
+        # create .conf file and add section and option to it
+        mkdir -p /etc/sssd
```

Comment:
It seems that this is a problem for the unit tests, I would try the similar trick as in the other macro (line 1831).